### ISSUE TITLE
Accept DELETE and PUT verbs for MVC6 when using Microsoft.AspNet.Hosting

### ISSUE THREAD

**Issue by mikeesouth** [aspnet/Hosting#725]
Hi, I'm using ASP.NET Core RC1 and I'm having a problem with my development server not accepting DELETE or PUT verbs for MVC Controller / Web API.

I'm hosting my development server using Microsoft.AspNet.Hosting. I have tried both Kestrel and WebListener:
```
  "commands": {
    "web1": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.Kestrel --server.urls http://localhost:5000 --ASPNET_ENV development",
    "web2": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5000 --ASPNET_ENV development"
  }
```

Both of these works fine except for one thing: I cannot call my MVC Controller actions with DELETE or PUT. It is working fine for GET and POST but when I try a DELETE call I get 404.

I've found a lot of support on this online but they all point to accepting these verbs in IIS and I don't want to use IIS for development. Here is one releveant link: [ASP.NET 5 + MVC 6 + Web API controller with PUT or PATCH verbs returns 404 error [closed]](http://stackoverflow.com/questions/28842587/asp-net-5-mvc-6-web-api-controller-with-put-or-patch-verbs-returns-404-error).

I've tried to call `app.UseIISPlatformHandler();´ in my Startup.cs and put a web.config in the root but it does not seem to be honored by Kestrel (or WebListener).

How can I configure Kestrel (or WebListener) to accept DELETE and PUT verbs?

**Comment by davidfowl** [comment]
What error do you get? You can try turning on logging in your application so you can see what happens to the request. I assume you can see the console output since you're not running through IIS right?

> I've tried to call `app.UseIISPlatformHandler();´ in my Startup.cs and put a web.config in the root but it does not seem to be honored by Kestrel (or WebListener).

If you're not using IIS, then this won't have any effect.

> How can I configure Kestrel (or WebListener) to accept DELETE and PUT verbs?

It should work fine by default. I'm not aware of any issues that would cause this to be blocked unless you are using IIS.

**Comment by mikeesouth** [comment]
Thanks for the quick answer. I was getting 404 (Not Found) for my DELETE actions on my controller. In light of your error I did however continue my troubleshooting and found the problem. It is now working with both Kestrel and WebListener. If anyone else runs into this problem the solution for me was that the HttpDelete attribute was specified like this:
```
[HttpDelete("{id}")]
public void Delete(int id)
```

And that did not work with my route config in Startup.cs:
```
routes.MapRoute(name: "default", template: "api/{controller=Controller}/{action=Index}/{id?}");
```

When I changed the attribute from `[HttpDelete("{id}")]` to just `[HttpDelete]` it started to work. I could then fire DELETE calls to `http://localhost:5000/api/Todo/Delete/3` to delete item with ID=3
